lib/gollum/app.rb

```diff
@@ -366,6 +366,9 @@ class App < Sinatra::Base
         wiki           = wiki_new
         @name          = params[:page] || "Preview"
         @page          = wiki.preview_page(@name, params[:content], params[:format])
+        ['sidebar', 'header', 'footer'].each do |subpage|
+          @page.send("set_#{subpage}".to_sym, params[subpage]) if params[subpage]
```

Comment:
What sets `params[subpage]`? Also, what does `params[subpage]` contain? A single string or an array of strings?

Reply:
It's being set by the ajax call [here](https://github.com/gollum/gollum/blob/5.x/lib/gollum/public/gollum/javascript/gollum.js.erb#L390). Each field contains a string.

Reply:
Got it!